src/domain/core/src/services/ingest/push_ingest_executor.rs

```diff
@@ -154,13 +154,16 @@ pub enum PushIngestError {
 
 pub enum DataSource {
     Url(url::Url),
+    Buffer(bytes::Bytes),
     Stream(Box<dyn AsyncRead + Send + Unpin>),
 }
 
+// TODO: Should this be Debug trait?
```

Comment:
It looks like yes. Only in the case of `url` we should take into account the fact that `Debug` outputs the whole (huge) structure there, not the value itself

Reply:
Changed to Debug to produce something like:
```
Url("https://url-as-string")
Bytes(...)
Stream(...)
```